src/System.Private.CoreLib/shared/System/MemoryExtensions.Fast.cs

```diff
@@ -141,9 +141,18 @@ public static int IndexOf(this ReadOnlySpan<char> span, ReadOnlySpan<char> value
                 return -1;
             }
 
+            if (comparisonType == StringComparison.Ordinal)
+            {
+                return SpanHelpers.IndexOf(
+                    ref MemoryMarshal.GetReference(span),
+                    span.Length,
+                    ref MemoryMarshal.GetReference(value),
+                    value.Length);
+            }
+
             if (GlobalizationMode.Invariant)
             {
-                return CompareInfo.InvariantIndexOf(span, value, string.GetCaseCompareOfComparisonCulture(comparisonType) != CompareOptions.None);
+                return CompareInfo.InvariantIndexOf(span, value, ignoreCase: true);
```

Comment:
This still needs to use `GetCaseCompareOfComparisonCulture(comparisonType) != CompareOptions.None`.